std/parallelism.d

```diff
@@ -945,11 +905,74 @@ if (is(typeof(fun(args))) && isSafeTask!F)
     return ret;
 }
 
+version(useSysctlbyname)
+    private extern(C) int sysctlbyname(
+        const char *, void *, size_t *, void *, size_t
+    ) @nogc nothrow;
+
 /**
 The total number of CPU cores available on the current machine, as reported by
 the operating system.
 */
-immutable uint totalCPUs;
+@property @nogc nothrow pure @trusted uint totalCPUs()
+{
+    return (cast(uint function() @nogc nothrow pure)
+        &totalCPUsImpl)();
+}
```

Comment:
Can't you use `lazilyInitializedConstant` for `totalCPUs`?

Reply:
ah, good idea brb